script.js

```diff
@@ -1,4 +1,259 @@
+  console.log(playerlists);
+  checkDiceForWinner();
+  if (mode == "reverse") {
+    // Game mode require us to print the lowest score first
+    for (let i = 0; i < playerlists.length; i++) {
+      output =
+        output +
+        "Playername: " +
+        playerlists[i].playername +
+        " " +
+        "Score is: " +
+        playerlists[i].score +
+        "<br>";
+    }
+  } else {
+    // Print highest score first
+    for (let i = playerlists.length - 1; i > -1; i--) {
+      console.log(i);
+      output =
+        output +
+        "Playername: " +
+        playerlists[i].playername +
+        " " +
+        "Score is: " +
+        playerlists[i].score +
+        "<br>";
+    }
+  }
+  console.log(output);
+  return output;
+}
+
+// function printScoreBoard(playerlists, mode) {
+//   var output = "The score board now!<br>";
+//   bblSort(plist); //sort the player list according to score lowest first
+//   console.log(playerlists);
+//   // Print lowest score first
+//   // if (mode == "reverse") {
+//   //   bblSortNumberLow(playerlists);
+//   // }
+//   checkDiceForWinner();
+//   bblSort(plist); //Sort player score highest to lowest
+//   for (let i = 0; i < playerlists.lenght; i++) {
+//     console.log(i);
+//     output =
+//       output +
+//       "Playername: " +
+//       playerlists[i].playername +
+//       " " +
+//       "Score is: " +
+//       playerlists[i].score +
+//       "<br>";
+//   }
+//   console.log(output);
+//   return output;
+// }
+
+const plist = []; // Array to keep all players data
+var inputPlayersAndDiceNumber = (inputGameMode = true);
+var startOfGame = true;
+var myOutputValue = "";
+
+var main = function (input) {
+  totalGames++;
+  // Player2_rolls.push(rollDice());
+  // console.log(Player2_rolls[0]);
+  //`Please enter the number of players and dices in the following format: Example 2 players and 2 dices, you will enter '2 2' in the input box`;
+
+  if (startOfGame) {
+    if (inputPlayersAndDiceNumber) {
+      numOfPlayers = input.split(" ")[0];
+      numOfDices = input.split(" ")[1];
+      myOutputValue = `Please enter your game mode: Input 'normal' or 'reverse' `;
+      console.log("numOfPlayers = " + numOfPlayers + " dice = " + numOfDices);
+      inputPlayersAndDiceNumber = false;
+    } else if (inputGameMode) {
+      gameMode = input;
+      inputGameMode = false;
+      console.log("inputGameMode " + gameMode);
+      myOutputValue = `Please enter your if you wish to autocombine dicenumber to the highest or lowest: Input 'high or 'low'`;
+    } else {
+      console.log("inputGameMode " + gameMode);
+      autoCombineOption = input;
+      startOfGame = false;
+      myOutputValue = `Game start.<br> Number of players = ${numOfPlayers}. Number of dices = ${numOfDices}. <br> Game Mode is ${gameMode}Click submit to start!`;
+      playersCreation(numOfPlayers); //All players data created will be sorted in plist array
+    }

```

Comment:
this is good but do play around with html and DOM manipulation in the future so that these modes can be entered using alternative means rather than free text input which would require input validation 

one way would be using additional buttons and input types